### PR TITLE
Code cleanup for the user account page

### PR DESCRIPTION
**Pull request by nicolasbrossard**
Removed the isErrorX calls (exceptions are now used).
Removed the unnecessary isset tests on $additionalUserInfo: it is always set.
Moved code in form_edit_user.tpl that scrolls the form to its first invalid field so that other forms can also benefit from this.
This is for Redmine ticket #7497

### CODE REVIEW

File: htdocs/js/invalid_form_scroll.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            // Only make the uppermost invalid element visible when the
+            // form is submitted
+            if(!bindInvalidFormListeners.scrollingDone) {
+                this.scrollIntoView(true);
+                // scrollingIntoView is not enough: the navigation bar will appear
+                // over the invalid element and hide it.
+                // We have to scroll an additional number of pixels down so that 
+                // the elements becomes visible. 
+                if(navbarHeader) {
+                    window.scrollBy(0,- $(navbarHeader).height()- 10);
```
Comment (samirdas) [inline htdocs/js/invalid_form_scroll.js]: Is this valid?

File: modules/user_accounts/php/NDB_Form_user_accounts.class.inc [hunk 2]
```diff
@@ -1,48 +1,82 @@
...
+                if ($editor->hasPermission('user_accounts_multisite')) {
+                    return true;
                 }
+
+                if (empty($this->identifier)) {
+                    return true;
+                }
+
+                return $editor->getData('CenterID') == $user->getData('CenterID');
             }
```
Comment (samirdas) [inline modules/user_accounts/php/NDB_Form_user_accounts.class.inc]: Check bracket for logic of if Empty identifier.

Comment (nicolasbrossard) [inline modules/user_accounts/php/NDB_Form_user_accounts.class.inc]: The original return statement, before my pull request was like this:

            return ($editor->hasPermission('user_accounts') &&  ($editor->hasPermission('user_accounts_multisite') || (empty($this->identifier) || $editor->getData('CenterID') == $user->getData('CenterID'))));

I just split it into multiple returns and I think it boils down to the exact same logic. If you do not think so, please provide counter-example.

File: htdocs/js/invalid_form_scroll.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                }
```
Comment (xlecours) [inline htdocs/js/invalid_form_scroll.js]: Yes. The scrollBy function takes 2 integers parameters. The second one is negative to scroll up by the height of the LORIS header menu - 10px.
----------
Syntax
window.scrollBy(X, Y);
Parameters

X is the offset in pixels to scroll horizontally.
Y is the offset in pixels to scroll vertically.
Positive co-ordinates will scroll to the right and down the page. Negative values will scroll to the left and up the page.

File: htdocs/js/invalid_form_scroll.js (hunk 1 above)
Comment (xlecours) [inline htdocs/js/invalid_form_scroll.js]: Yes. The scrollBy function takes 2 integers parameters. The second one is negative to scroll up by the height of the LORIS header menu - 10px.

Syntax
window.scrollBy(X, Y);
Parameters

X is the offset in pixels to scroll horizontally.
Y is the offset in pixels to scroll vertically.
Positive co-ordinates will scroll to the right and down the page. Negative values will scroll to the left and up the page.

File: modules/user_accounts/php/NDB_Form_user_accounts.class.inc (hunk 2 above)
Comment (xlecours) [inline modules/user_accounts/php/NDB_Form_user_accounts.class.inc]: It is exactly the same.
we have 4 conditions A,B,C and D. 
Before:  A and ( B or ( C or D ) )
Now: A and ( B or C or D )

The last return could have be righten like this to keep the format : 
if ( $editor->getData('CenterID') == $user->getData('CenterID') ) {
    return true;
}

**Pull request merged**